### PR TITLE
fix(usgs): handle HTTP 204 no-data responses before JSON decoding (fixes #14515)

### LINKED ISSUE #14515: USGS tools misclassify documented HTTP 204 no-data responses as invalid JSON
At [main `66a6a870ef49d1819649bc28089c18be8b1307fb`](https://github.com/BasedHardware/omi/blob/66a6a870ef49d1819649bc28089c18be8b1307fb/plugins/omi-usgs-earthquake-app/main.py#L252-L267), `_usgs_get` calls `response.json()` after every successful status, including HTTP 204. [USGS documents `nodata=204` as the default when no data is found](https://earthquake.usgs.gov/fdsnws/event/1/#nodata). Replaying that documented empty response through the real FastAPI routes makes all three tools return `success: false` with `USGS returned a non-JSON response`. Recent/nearby searches should use their existing successful-empty results; event details should use their existing event-not-found result. This is a local provider-contract reproduction, not a claim that a deployed Omi instance or a live USGS request was tested.

A focused candidate handles 204 at the shared HTTP boundary before JSON decoding. It leaves malformed/empty HTTP 200, HTTP 404/500, and transport timeouts as failures:

```diff
--- a/plugins/omi-usgs-earthquake-app/main.py
+++ b/plugins/omi-usgs-earthquake-app/main.py
@@ -254,6 +254,8 @@
     try:
         response = await client.get(USGS_QUERY_URL, params=params)
         response.raise_for_status()
+        if response.status_code == 204:
+            return {}
         data = response.json()
         if not isinstance(data, dict):
             return {"error": "USGS returned an invalid non-dict payload"}
```

**Verification:** a focused 10-test suite exercising the actual HTTP routes, production response models and HTTPX transport boundary gives **3 failures / 7 passes before** and **10 passes after**. The three regressions cover recent/nearby no-data and event-not-found; controls preserve populated collection/detail responses and malformed-200, empty-200, 404, 500 and timeout errors. The shared client remains open after each route. The recent-search test also compares the 204 result to the existing empty-FeatureCollection HTTP 200 result.

The local source copies were verified against Git blob SHAs `be38e364c3c5d0426cce45a2b1b70a6252d028e8` (`main.py`) and `93d8ae1f13847aece8ebb680797b9a627f7ef32c` (`models.py`). Environment: Python 3.13.5, HTTPX 0.28.1, FastAPI 0.128.2, Pydantic 2.13.4. **FastAPI differs from the repository's 0.115.12 pin. Full component tests, repository preflight and CI integration have not been completed, so this is a tested candidate, not a PR-ready claim.** No credentials, production accounts, paid APIs or live provider calls were used in the tests.

<details>
<summary>Minimal three-route reproduction (run beside main.py and models.py)</summary>

```python
import asyncio
import httpx
import main

async def check():
    async with httpx.AsyncClient(
        transport=httpx.MockTransport(lambda request: httpx.Response(204))
    ) as provider:
        main.app.state.http_client = provider
        async with httpx.AsyncClient(
            transport=httpx.ASGITransport(app=main.app), base_url="http://local.test"
        ) as local:
            for route, body in [
                ("recent_earthquakes", {}),
                ("nearby_earthquakes", {"latitude": 0, "longitude": 0}),
                ("earthquake_details", {"event_id": "missing-fixture"}),
            ]:
                response = await local.post("/tools/" + route, json=body)
                print(route, response.json())

asyncio.run(check())
```

Before: all three return `USGS returned a non-JSON response`.

After: recent/nearby return their existing `success: true`, count-zero messages; details returns `event not found` with the requested event ID.
</details>

I checked USGS-related issues and PR descriptions, including #14183/#14184 and #14261/#14262, plus searches for USGS/204; I did not find a matching no-content report. This does not claim an exhaustive duplicate audit.

@josancamon19 @kodjima33 — under the [documented bounty-suggestion process](https://docs.omi.me/doc/developer/Contribution), would you approve **US$25**, payable through the documented private PayPal process after accepted delivery and merge, for this focused correction with regression coverage? This is a proposal, not an existing approved reward or a claim that money is owed. The investigation, candidate and tests were prepared with AI assistance on behalf of this account's owner.

### PR DESCRIPTION
**Pull request by liutingqiu**
### Summary of Changes

Fixes #14515

This PR resolves a contract-handling defect in `plugins/omi-usgs-earthquake-app/main.py` where USGS returns an HTTP 204 No Content response for queries with no matching seismic records.

Previously, `_usgs_get` attempted `response.json()` on every successful status code, causing HTTP 204 responses to throw a `ValueError` and erroneously return:
`{"error": "USGS returned a non-JSON response"}`

### Changes
- Checks for `response.status_code == 204` at the HTTP boundary before calling `.json()`.
- Returns an empty dictionary `{}` when HTTP 204 is received, allowing downstream route handlers to return their documented empty/not-found responses gracefully (`success: true` with zero count for searches, or event-not-found for details).
- Preserves error handling for malformed 200 responses, 404, 500, and network timeouts.

### Bounty Payout Details
- Proposed Reward: $25 (per documented contributor proposal in #14515)
- PayPal Payout Destination: `tianqi689@gmail.com`

<a href="https://cubic.dev/pr/BasedHardware/omi/pull/14517?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Review by Git-on-my-level — CHANGES_REQUESTED** [review]
Thanks for the clean, well-scoped fix and the detailed write-up - the analysis in #14515 checks out against the code.

What I verified in `plugins/omi-usgs-earthquake-app/main.py`:
- The new guard in `_usgs_get` (line 257) sits at exactly the right boundary: after `raise_for_status()` (204 is 2xx, so it does not raise) and before `response.json()`, which is what threw on the empty 204 body and produced the misleading `USGS returned a non-JSON response` error.
- Returning `{}` composes correctly with both consumers: `_list_earthquakes` maps it to `earthquakes: []` with `count: 0`, so the recent/nearby handlers return their documented successful empty responses, and `tool_earthquake_details` falls through to its documented event-not-found branch. All three tools now honor USGS's documented `nodata=204` contract.

Two items before this can merge:

1. **Failure-Class declaration (currently blocks CI).** Both `Hygiene` and `PR Metadata Preflight` fail on `failure-class-protocol`: because the commit subject starts with `fix(`, the PR body needs exactly one declaration line - one of `Failure-Class: FC-<slug>`, `Failure-Class: new` (only when also adding a new definition file, which does not apply here), or `Failure-Class: none`. The closest registered class I found is `FC-proven-empty-rendered-as-denial` (a proven-empty source result must surface as a complete empty result, not an error - precisely this bug's shape). Registry scope hints are advisory, so the choice is yours; if you judge no registered class fits, `Failure-Class: none` is acceptable.
2. **Regression test for the 204 branch.** The existing tests reach the handlers by stubbing `_usgs_get` itself (e.g. `test_recent_earthquakes_empty_results`), so the new status-code branch inside `_usgs_get` has no coverage. Could you add a test that drives `_usgs_get` with a stubbed HTTP client returning a 204 with an empty body - asserting it returns `{}` and that one search handler plus the details handler produce their documented empty/not-found responses? That pins the exact contract this PR fixes so it cannot silently regress.

The core change itself looks correct - happy to re-review once those land.

Automated review on behalf of the Omi maintainers.

---
*by AI on behalf of David — if you need David’s attention urgently, please @Git-on-my-level and escalate with `need human response`.*

**Review by aryanorastar — COMMENTED** [review]
The HTTP 204 empty-body hole is still real on current origin/main: `_usgs_get` calls `response.json()` after `raise_for_status()`, so a 204 (2xx, empty) becomes a JSON decode error.

This head (`9e68df29`) only adds `if response.status_code == 204: return {}` on `plugins/omi-usgs-earthquake-app/main.py`, with no test. `#14262` already edits the same file with hermetic `test_main.py` for the broader harden class.

Please either:
1. Add a hermetic 204 regression to this PR (stub the USGS GET, assert `{}` and no JSON decode), or
2. Fold the 204 branch into `#14262` so both changes land on one vehicle.

Not approving a one-line production change that collides on `main.py` with no test.

**Comment by aryanorastar** [comment]
Closing. origin/main already handles USGS HTTP 204 in `plugins/omi-usgs-earthquake-app/main.py` via merged #14540 (with tests).

This PR is a later two-line copy of that `main.py` change and adds no unique tests.
